### PR TITLE
Unfinishedsenten patch 2

### PR DESCRIPTION
**Pull request by unfinishedsenten**
-Changed Appointment HTML Layout
-Changed appointment Creation HTML layout
-Implemented ThymeLeaf integration
-Modified appointment controller
-Added 'time' var for Appointment.java
-Added "/appointment" for SecurityConfiguration.java
- Added working button 'Home' to navigate to /notifications
- Modified newAppointmentDTO.java for future backend and revision to work in conjunction with Appointment.java

### CODE REVIEW

File: src/main/java/com/virtudoc/web/controller/appointmentController.java [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
+
+		return "appointment";
 	}
-	@GetMapping("/submit")
-	String getSubmit() {
-		return "confirmation";
+
+	@PostMapping("/appointment")
+	public String submitForm(@ModelAttribute("appointment") Appointment appointment) {
+		System.out.println(appointment);
```
Comment (ARMmaster17) [inline src/main/java/com/virtudoc/web/controller/appointmentController.java:41]: Appointment data could theoretically contain PHI. To maintain HIPAA compliance, might be worth removing this line once you're done debugging the Appointments controller.

**Review by ARMmaster17 — APPROVED** [review]

File: src/main/java/com/virtudoc/web/controller/appointmentController.java (hunk 1 above)
Comment (unfinishedsenten) [reply src/main/java/com/virtudoc/web/controller/appointmentController.java:41]: Are you talking about line 41? I was testing something from a tutorial and forgot to delete it. I can remove that.

Comment (ARMmaster17) [reply src/main/java/com/virtudoc/web/controller/appointmentController.java:41]: Yeah, just L41. The rest is good.

**Comment by unfinishedsenten** [comment]
> ```
> org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; constraint [time" of relation "appointments]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement
> 	at com.virtudoc.web.AppointmentRepositoryTests.testAddNew(AppointmentRepositoryTests.java:30)
> Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
> 	at com.virtudoc.web.AppointmentRepositoryTests.testAddNew(AppointmentRepositoryTests.java:30)
> Caused by: org.postgresql.util.PSQLException: 
> ERROR: null value in column "time" of relation "appointments" violates not-null constraint
>   Detail: Failing row contains (1, Feb 3, Doctor 1, test@test.com, Kennesaw, Patient 1, Reason for Visit, Covid, null).
> 	at com.virtudoc.web.AppointmentRepositoryTests.testAddNew(AppointmentRepositoryTests.java:30)
> ```
> 
> Looks like since you added a new field with a constraint to the Appointment entity the tests need to be updated to include the new field in `AppointmentRepositoryTests.java`. Otherwise you can delete the constraint and either way the tests should pass.

Yes, I was looking at this now, I added the time constraint for the entity class and also added 'apt.setTime("01:30 PM"); ' as a test case but it still fails. It's a string, so it should pass through... but still is not.

**Comment by unfinishedsenten** [comment]
Fixed the test failure. Line 10, Appointment.java 
@GeneratedValue(strategy = GenerationType.IDENTITY) 
to
@GeneratedValue(strategy = GenerationType.AUTO)

**Pull request merged**